service/src/main/java/com/codeforcommunity/processor/BlocksProcessorImpl.java

```diff
@@ -109,6 +114,36 @@ public BlockResponse resetBlocks(JWTData jwtData, List<String> blockIds) {
         .collect(Collectors.toList());
   }
 
+  @Override
+  public GetAssignedBlocksResponse getAllReservedBlocks(JWTData jwtData) {
+    return getAssignedBlocksWithStatus(jwtData, BlockStatus.RESERVED);
+  }
+
+  @Override
+  public GetAssignedBlocksResponse getAllDoneBlocks(JWTData jwtData) {
+    return getAssignedBlocksWithStatus(jwtData, BlockStatus.DONE);
+  }
+
+  private GetAssignedBlocksResponse getAssignedBlocksWithStatus(
+      JWTData jwtData, BlockStatus status) {
+    if (jwtData.getPrivilegeLevel() != PrivilegeLevel.ADMIN) {
```

Comment:
I think this check should be taken out of this helper method. It doesn't seem to match up with the purpose of this method and would allow us to not pass the JWTData